### DISCUSSION TITLE
Old Endpoint?

### DISCUSSION THREAD

**Discussion by IsaacGemal** [microsoft/AI-Chat-App-Hack#98] [Ideas]
Hi there.

Just something I noticed in the azure openai studio - I checked out the sample code and it looks like it has the old openai endpoint
![chrome_tXM1Jq2GYs](https://github.com/microsoft/AI-Chat-App-Hack/assets/147355120/e5c866ba-b863-4be0-ac56-88b96500c145)

I believe that `completion = openai.ChatCompletion.create(` should become `completion = client.chat.completions.create`

Per [this ](https://platform.openai.com/docs/api-reference/chat/create)link

Something like this?

```python
from openai import OpenAI

client = OpenAI()

# Assuming deployment_id and other Azure specific variables are defined elsewhere
messages = [
    {"role": "system", "content": "You are a helpful assistant."},
    {"role": "user", "content": "What are the differences between Azure Machine Learning and Azure AI services?"}
]

completion = client.chat.completions.create(
    model="gpt-3.5-turbo",
    messages=messages,
    data_sources=[  # Adjusted to new format, maintaining your AzureCognitiveSearch config
        {
            "type": "AzureCognitiveSearch",
            "parameters": {
                "endpoint": "$search_endpoint",
                "indexName": "$search_index",
                "semanticConfiguration": None,  # Python style for null values
                "queryType": "vectorSimpleHybrid",
                "fieldsMapping": {},
                "inScope": True,
                "roleInformation": "You are an AI assistant that helps people find information.",
                "filter": None,
                "strictness": 3,
                "topNDocuments": 5,
                "key": "$search_key",
                "embeddingDeploymentName": "ada-embedding-model"
            }
        }
    ],
    temperature=0,
    top_p=1,
    max_tokens=800,
    stop=None,
    stream=True
)

print(completion.choices[0].message)

```
The code I just pasted is straight from GPT-4 asking it to update the existing code, so treat it with caution.

**Comment by pamelafox** [comment]
Yes, that's true! I'll nudge the team and ask when they'll be updating to the 1.x series code. You would also need to upgrade the client to AzureOpenAI for user with Azure. See https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/switching-endpoints
